### PR TITLE
RUB-6 Created, Level model

### PR DESCRIPTION
**Pull request by sonishc**
**Reviewers:**
- [x] @rokor 
- [x] @mayar2610 
- [x] @vlodkopits
**Description:**
Fixed bug. Rebase done. Used straightforward name for variables in Level model.

### CODE REVIEW

File: app/models/level.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Level
+class Level < ApplicationRecord
+  # has_many :users | need when connecting with the User model
+  # Explain (  level.id is a user_level  )
+
+  validates_presence_of :level_experience, :level_health_point
```
Comment (rokor) [inline app/models/level.rb]: please use more straightforward name, for example not `level_experience, level_health_point` but `experience_level, health_point_level`

File: app/models/level.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  validates :level_experience, :level_health_point,
+            numericality: { only_integer: true },
+            uniqueness: true
+  validates :level_experience, :level_health_point, length: { maximum: 5 }
+  max_level = 12
```
Comment (rokor) [inline app/models/level.rb]: should be const.

**Review by rokor — CHANGES_REQUESTED** [review]

**Review by mayar2610 — APPROVED** [review]

**Review by vlodkopits — APPROVED** [review]

**Review by rokor — APPROVED** [review]

**Comment by rokor** [comment]
👍

**Comment by rokor** [comment]
https://www.devroom.io/2011/07/05/git-squash-your-latests-commits-into-one/

File: app/models/level.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Level
```
Comment (okohuch) [inline app/models/level.rb]: remove comment

File: app/models/level.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+class Level < ApplicationRecord
+  # has_many :users | need when connecting with the User model
+  # Explain (  level.id is a user_level  )
```
Comment (okohuch) [inline app/models/level.rb]: bad practice to write the code in advance if you do not need it not

Comment (okohuch) [reply app/models/level.rb]: remove these comments

File: app/models/level.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  validates :experience_level, :health_point_level,
+            numericality: { only_integer: true },
+            uniqueness: true
+  validates :experience_level, :health_point_level, length: { maximum: 5 }
+  MAX_LEVEL = 12
+  validates_inclusion_of :id, in: 1..MAX_LEVEL
+  validates_inclusion_of :experience_level, in: 50..10_000
+  validates_inclusion_of :health_point_level, in: 100..10_000
+
+  # validates_associated :users | need when connecting with the User model
```
Comment (okohuch) [inline app/models/level.rb]: remove comment

File: app/models/level.rb (hunk 3 above)
Comment (sonishc) [reply app/models/level.rb]: fixed

File: app/models/level.rb (hunk 4 above)
Comment (sonishc) [reply app/models/level.rb]: fixed.

File: app/models/level.rb (hunk 5 above)
Comment (sonishc) [reply app/models/level.rb]: fixed

File: spec/models/level_spec.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
+RSpec.describe Level, type: :model do
+  it 'has a valid Factory' do
+    build(:level).should be_valid
+  end
+  it { is_expected.to validate_presence_of(:experience_level) }
+  it { is_expected.to validate_presence_of(:health_point_level) }
+end
```
Comment (viktoriia-bilous) [inline spec/models/level_spec.rb]: There are no tests for invalid cases.

File: db/migrate/20170925134115_create_levels.rb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# CreateLevels
```
Comment (rokor) [inline db/migrate/20170925134115_create_levels.rb]: remove comment and check if migration currently working.

File: db/schema.rb [hunk 8]
```diff
@@ -11,7 +11,7 @@
 # It's strongly recommended that you check this file into your version control system.
 
 ActiveRecord::Schema.define(version: 20170927110812) do
-
+
```
Comment (rokor) [inline db/schema.rb:14]: not a fun of committing schema in PR. Also rebase needed.

File: spec/support/factory_girl.rb [hunk 9]
```diff
@@ -0,0 +1,22 @@
+RSpec.configure do |config|
```
Comment (rokor) [inline spec/support/factory_girl.rb]: It is look like this config is already in develop.Please rebase.

**Review by rokor — CHANGES_REQUESTED** [review]

**Pull request closed without merging**